llvm/lib/Transforms/Scalar/NewGVN.cpp

```diff
@@ -2666,6 +2671,7 @@ Value *NewGVN::findLeaderForInst(Instruction *TransInst,
   TempToBlock.insert({TransInst, PredBB});
   InstrDFS.insert({TransInst, IDFSNum});
 
+  PhiOfOpsCandidate = TransInst;
```

Comment:
This never gets reset -- isn't it going to leak into all future calls of performSymbolicEvaluation?

Reply:
You are right :) This is set every time that we process one of the operands of the new phi node. After that, I do not initialize to nullptr. Anyway, it is not needed any more.